microsoft/testsuites/core/lsvmbus.py

```diff
@@ -104,3 +107,149 @@ def lsvmbus_count_devices_channels(self, node: Node) -> None:
+
+    @TestCaseMetadata(
+        description="""
+        This test case will
+            1. Looks for the VMBus heartbeat device properties.
+            2. Checks the properties can be read and that the folder structure exists.
+            3. Checks that the in_* files are equal to the out_* files
+             when read together.
+            4. Checks the interrupts and events values are increasing during
+             reading them.
+        """,
+        priority=4,
+        requirement=simple_requirement(
+            supported_platform_type=[AZURE],
+        ),
+    )
+    def verify_vmbus_heartbeat_properties(self, node: Node) -> None:
+        lsvmbus_tool = node.tools[Lsvmbus]
+        vmbus_devices_list = lsvmbus_tool.get_device_channels()
+        vmbus_heartbeat_device = [
+            x for x in vmbus_devices_list if x.name == "Heartbeat"
+        ]
+        assert vmbus_heartbeat_device, "no vmbus heartbeat device"
+        vmbus_heartbeat_device = vmbus_heartbeat_device[0]
+        device_path = f"/sys/bus/vmbus/devices/{vmbus_heartbeat_device.device_id}"
+        node.execute(
+            f"ls -lt {device_path}",
+            sudo=True,
+            shell=True,
+            expected_exit_code=0,
+            expected_exit_code_failure_message=f"{device_path} doesn't exist",
+        )
+        vmbus_driver_files = [
+            "channel_vp_mapping",
+            "class_id",
+            "device",
+            "device_id",
+            "id",
+            "in_intr_mask",
+            "in_read_bytes_avail",
+            "in_read_index",
+            "in_write_bytes_avail",
+            "in_write_index",
+            "modalias",
+            "out_intr_mask",
+            "out_read_bytes_avail",
+            "out_read_index",
+            "out_write_bytes_avail",
+            "out_write_index",
+            "state",
+            "uevent",
+            "vendor",
+        ]
+        vmbus_channel_files = [
+            "cpu",
+            "events",
+            "in_mask",
+            "interrupts",
+            "out_mask",
+            "read_avail",
+            "write_avail",
+        ]
+        cmd_result = node.execute(
+            f"ls -lt {device_path}/driver_override",
+            sudo=True,
+            shell=True,
+        )
+        # driver_override file not existing - contain recent commits
+        if 0 != cmd_result.exit_code:
+            vmbus_driver_files.append(
+                [
+                    "monitor_id",
+                    "server_monitor_conn_id",
+                    "server_monitor_latency",
+                    "server_monitor_pending",
+                    "client_monitor_conn_id",
+                    "client_monitor_latency",
+                    "client_monitor_pending",
+                ]
+            )
+            vmbus_channel_files.append(["latency", "pending"])
+        for file in vmbus_driver_files:
+            node.execute(
+                f"ls -lt {device_path}/{file}",
```

Comment:
Can we add a tool `Ls`, which can be used to `list`, `exists`, `is_dir`?

Reply:
added